builder/dockerfile/dispatchers.go

```diff
@@ -769,6 +780,10 @@ func arg(b *Builder, args []string, attributes map[string]bool, original string)
 		b.options.BuildArgs[name] = &newValue
 	}
 
+	// Arg before FROM doesn't add layers
+	if b.image == "" {
+		return nil
+	}
 	return b.commit("", b.runConfig.Cmd, fmt.Sprintf("ARG %s", arg))
```

Comment:
I see that something like this was discussed already in #29766. I think we could go with either approach.